chaosinstana/api.py

```diff
@@ -0,0 +1,66 @@
+# -*- coding: utf-8 -*-
+from typing import Any, Dict
+
+from chaoslib.types import Configuration, Secrets
+from logzero import logger
+import requests
+
+
+__all__ = ['call_api', 'get_all_events']
+
+
+def call_api(path: str, configuration: Configuration,
+             secrets: Secrets, query: Dict[str, Any] = None,
+             body: Dict[str, Any] = None, method: str = "GET",
+             headers: Dict[str, str] = None) -> requests.Response:
+    """
+    TBD
+    """
+    # location given here
+    URL = "https://httpbin.org/get"
+
+    # defining a params dict for the parameters to be sent to the API
+    # PARAMS = {'address':location}
+
+    # sending get request and saving the response as response object
+    r = requests.get(url=URL)
+    # extracting data in json format
+    data = r.json()
+
+    return data
+
+
+def get_all_events(from_time: str, to_time: str,
+                   configuration: Configuration, secrets: Secrets) -> str:
+    """
+     get all events from instana within a time window, given by the from_time
+     and the to_time for details of the api see
+     https://instana.github.io/openapi/#tag/Events
+    """
+    instana_host = configuration.get("instana_host")
+    instana_api_token = secrets.get("instana_api_token")
+
+    if (instana_host and instana_api_token):
+
+        url = "{}/api/events".format(
+            configuration.get("instana_host"))
+
+        params = {}
+        if (from_time):
+            params["from"] = from_time
+
+        if (to_time):
+            params["to"] = to_time
+
+        auth_header_key = "authorization"
+        auth_header_contents = "apiToken " + secrets.get("instana_api_token")
+
+        r = requests.get(url=url,
+                         params=params,
+                         headers={auth_header_key: auth_header_contents})
+
+        data = r.json()
+
+        return data
+    else:
```

Comment:
That should be at the start of the function for readability:

```
from chaoslib.exceptions import ActivityFailed

if not (instana_host and instana_api_token):
    raises ActivityFailed("No Instana Host or API Token Secrete were found.")

...
```

Also, please make the message more actionable for the user :) Where should the add them when missing?